packages/lesswrong/components/posts/PostsEditForm.tsx

```diff
@@ -67,11 +69,21 @@ const PostsEditForm = ({ documentId, classes }: {
     extraVariablesValues: { eventForm: document?.isEvent }
   });
   const rateLimitNextAbleToPost = userWithRateLimit?.rateLimitNextAbleToPost
-  
+
   if (!document && loading) {
     return <Components.Loading/>
   }
 
+  // We auto-archive dialogues that have gone stale
+  const twoWeeksAgo = new Date();
+  twoWeeksAgo.setDate(twoWeeksAgo.getDate() - 14);
+  const lastRevisionAt = document?.lastRevisionAt
+  const potentialDialogue = isLWorAF && document?.collabEditorDialogue && document?.draft && !document?.deletedDraft
+  const dialogueAutoArchiveNotice = potentialDialogue && !!lastRevisionAt && new Date(lastRevisionAt) < twoWeeksAgo
+  const daysLeftUntilAutoArchive = moment(lastRevisionAt).add(autoArchiveCutoffDays, 'days').diff(moment.utc(), 'days')
```

Comment:
If you're going to use `moment`, prefer to use it consistently, both so that it's more readable and so that you don't trip up on inconsistencies between `moment` and native JS date manipulation